mir_eval/alignment.py

```diff
@@ -0,0 +1,300 @@
+        estimated timestamps, in seconds
+    duration : float
+        Total duration of audio (seconds) - this is needed to compute the final segments properly!
+    Returns
+    -------
+    pcs : float
+        Percentage of time where ground truth and predicted segments overlap
+    """
+    validate(reference_timestamps, estimated_timestamps)
+    duration = float(duration)
+    assert np.max(reference_timestamps) <= duration, (
+        "Expected largest reference timestamp "
+        f"{np.max(reference_timestamps)} to not be "
+        f"larger than duration {duration}"
+    )
+    assert np.max(estimated_timestamps) <= duration, (
+        "Expected largest estimate timestamp "
+        f"{np.max(estimated_timestamps)} to not be "
+        f"larger than duration {duration}"
+    )
+
+    ref_starts = np.concatenate([[0], reference_timestamps])
+    ref_ends = np.concatenate([reference_timestamps, [duration]])
+    est_starts = np.concatenate([[0], estimated_timestamps])
+    est_ends = np.concatenate([estimated_timestamps, [duration]])
+    overlap_starts = np.maximum(ref_starts, est_starts)
+    overlap_ends = np.minimum(ref_ends, est_ends)
+    overlap_duration = np.sum(np.maximum(overlap_ends - overlap_starts, 0))
+    return overlap_duration / duration
+
+
+def perceptual_synchrony(offset):
+    """function perceptually weighting the deviations of onsets"""
+    skewness = 1.12244251
+    localisation = -0.22270315
+    scale = 0.29779424
+    normalisation_factor = 1.6857
+    return (1.0 / normalisation_factor) * skewnorm.pdf(
+        offset, skewness, loc=localisation, scale=scale
+    )
+
+
+def perceptual_metric(reference_timestamps, estimated_timestamps):
+    """metric based on human synchronicity perception as measured in the paper
+
+    "User-centered evaluation of lyrics to audio alignment",
+    N. Lizé-Masclef, A. Vaglio, M. Moussallam, ISMIR 2021
+
+    The parameters of this function were tuned on data collected through a user Karaoke-like
+    experiment
+    It reflects human judgment of how "synchronous" lyrics and audio stimuli are perceived
+    in that setup.
+    Beware that this metric is non-symmetrical and by construction it is also not equal to 1 at 0.
+
+    Examples
+    --------
+    >>> reference_timestamps = mir_eval.io.load_events('reference.txt')
+    >>> estimated_timestamps = mir_eval.io.load_events('estimated.txt')
+    >>> perceptual = mir_eval.align.perceptual_metric(reference_onsets, estimated_timestamps)
+
+    Parameters
+    ----------
+    reference_timestamps : np.ndarray
+        reference timestamps, in seconds
+    estimated_timestamps : np.ndarray
+        estimated timestamps, in seconds
+    Returns
+    -------
+    perceptual_score : float
+        Perceptual score, averaged over all timestamps
+    """
+    validate(reference_timestamps, estimated_timestamps)
+    perceptual_score = perceptual_synchrony(estimated_timestamps - reference_timestamps)
```

Comment:
Does this really need its own function if this is the only place it's called?

Reply:
Probably not, integrated it into the main function which was very empty anyway